indico/modules/rb/schemas.py

```diff
@@ -394,6 +397,23 @@ def _check_weekdays_unique(self, weekdays, **kwargs):
         if weekdays and len(weekdays) != len(set(weekdays)):
             raise ValidationError('Duplicate weekdays')
 
+    @validates_schema(skip_on_field_errors=True)
+    def _check_booking_reason(self, data, **kwargs):
+        link = data.get('link_id')
+        if not link:
+            booking = self.context.get('booking')
+            if booking:
+                link = booking.link_id
```

Comment:
I don't think this should try both (passing a `link_id` during editing would be wrong). If there's a booking in the context, then that should (only) be used, and otherwise it should use the `link_id` from data.

Reply:
```python
booking = self.context['booking']
link_id = booking.link_id if booking else data.get('link_id')
```

This should do the job (and is much more readable).

And below you can then do `validate = link_id is None`

Reply:
Good suggestion! Just changed it.